app/gui/view/graph-editor/src/component/node/input/widget.rs

```diff
@@ -146,6 +168,25 @@ pub trait SpanWidget {
     }
 }
 
+/// Description of how well a widget matches given node. Used to determine which widget should be
+/// used, or whether the applied widget override is valid in given context.
+#[derive(Debug, Default, Clone, Copy, PartialOrd, Ord, PartialEq, Eq)]
+pub enum Score {
+    /// This widget kind cannot accept the node. It will never be used, even if it was explicitly
+    /// requested using an override.
+    NotMatched,
+    /// A bad, but syntactically valid match. Matching widget kind will only be used if it was
+    /// explicitly requested using an override. Should be the default choice for cases where
+    /// the node is syntactically valid in this widget's context, but no sensible defaults can
+    /// be inferred from context.
+    #[default]
+    OnlyOverride,
+    /// A good match, but there might be a better one. Use this widget if there is no better
+    /// option.
+    Good,
+    /// Widget matches perfectly, stop searching and use this widget immediately.
+    Perfect,
```

Comment:
```suggestion
    /// A good match, but there might be a better one. This widget will be used if there is no better
    /// option.
    Good,
    /// Widget matches perfectly, and be taken outright, without checking other kinds.
    Perfect,
```
Just a proposition. The previous variants don't speak to me, as if it would be _me_ the one searching for widget kind.